docs/CONVERTING_TO_HOMED.md

```diff
@@ -0,0 +1,137 @@
+* [JSON User Records](https://systemd.io/USER_RECORD)
+* [JSON Group Records](https://systemd.io/GROUP_RECORD)
+* [User/Group Record Lookup API via Varlink](https://systemd.io/USER_GROUP_API)
+
+## Caveat
+
+This is a manual process, and possibly a bit fragile. Hence, do this at your
+own risk, read up beforehand, and make a backup first. You know what's at
+stake: your own home directory, i.e. all your personal data.
+
+## Step-By-Step
+
+Here's the step-by-step guide:
+
+0. Make sure to backup your home directory, and make sure to log out of your
+   user account fully. Make sure you run a distribution that has
+   `systemd-homed` enabled and properly set up, including the necessary PAM and
+   NSS configuration updates. Make sure you have enough disk space in `/home/`
+   for a (temporary) second copy of your home directory.
+
+1. Log in as root on the console.
+
+2. Rename your existing home directory to something safe. Let's say your user
+   ID is `foobar`. Then do:
+
+    ```
+    mv /home/foobar /home/foobar.saved
+    ```
+
+3. Have a look at your existing user record, as stored in `/etc/passwd` and
+   related files. We want to use the same data for the new record, hence it's good
+   looking at the old data. Use commands such as:
+
+    ```
+    getent passwd foobar
+    getent shadow foobar
+    ```
+
+   This will tell you the `/etc/passwd` and `/etc/shadow` entries for your
+   user. For details about the fields, see the respective man pages
+   [passwd(5)](http://man7.org/linux/man-pages/man5/passwd.5.html) and
+   [shadow(5)](http://man7.org/linux/man-pages/man5/shadow.5.html).
+
+   The fourth field in the `getent passwd foobar` output tells you the GID of
+   your user's main group. Depending on your distribution it's a group private
+   to the user, or a group shared by most local, regular users. Let's say the
+   GID reported is 1000, let's then query its details:
+
+    ```
+    getent group 1000
+    ```
+
+   This will tell you the name of that group. If the name is the same as your
+   user name your distribution apparently provided you with a private group for
+   your user. If it doesn't match (and is something like `users`) it apparently
+   didn't. Note that `systemd-homed` will always manage a private group for
+   each user under the same name, hence if your distribution is one of the
+   latter kind, then there's a (minor) mismatch in structure when converting.
+
+   Save the information reported by these three commands somewhere, for later
+   reference.
+
+4. Now edit your `/etc/passwd` file and remove your existing record
+   (i.e. delete a single line, the one of your user's account, leaving all
+   other lines unmodified). Similar for `/etc/shadow`, `/etc/group` (in case
+   you have a private group for your user) and `/etc/gshadow`. Most
+   distributions provide you with a tool for that, that adds safe
+   synchronization for these changes: `vipw`, `vipw -s`, `vigr` and `vigr -s`.
```

Comment:
Why not just do `groupdel` and `userdel`? That's generally quicker and safer than editing files.

Reply:
Wel, userdel/groupdel are named differently on the various distros, and I didn't want to spend too much time on that. Also there's just too much opaque stuff behind this commands we don't want. For example, there's a callout script in Fedora's userdel which is supposed to remove at/cron scripts, and I am not sure we should do that. We really just want to remove the record itself, hence I figured telling the user to remove it directly is easiest, clearest and the least likely to have other effects that we do not want.

Reply:
> there's a callout script in Fedora's userdel which is supposed to remove at/cron scripts

Do you have any more information about this? I looked at shadow-utils (which provides the commands) and cronie-* packages, and I didn't see anything relevant. A quick tests with removing and recreating a user shows that user's crontab entries are preserved.

Reply:
`man 8 userdel`:

```
       USERDEL_CMD (string)
           If defined, this command is run when removing a user. It should
           remove any at/cron/print jobs etc. owned by the user to be removed
           (passed as the first argument).

           The return code of the script is not taken into account.

           Here is an example script, which removes the user's cron, at and
           print jobs:
```

Note sure Fedora actually makes use of this, but it's definitely something userdel knows and does...

Reply:
I'm pretty sure that this is something that almost nobody knows about. It is (or was, more likely) used on big installations. The number of *laptops*, which is what homed is targetting, using this is probably zero. OTOH, I think the chances that people screw up on of their passwd files when editing them by hand is >0. But let's close this thread.